containers/tenant-namespace-operator/roles/tenantnamespace/tasks/main.yml

```diff
@@ -157,6 +157,63 @@
   when:
   - loadBalancerIP is defined
 
+# Delete resources that have selectors that need to be updated
+- name: Remove upgrade resources
+  k8s:
+    state: absent
+    api_version: "{{ item.api_version }}"
+    kind: "{{ item.kind }}"
+    namespace: "{{ item.namespace }}"
+    name: "{{ item.name }}"
+  loop:
+  - api_version: apps/v1
+    kind: Deployment
+    namespace: "{{ meta.name }}-admin"
+    name: "{{ meta.name }}-ingress-controller"
+  - api_version: v1
+    kind: Service
+    namespace: "{{ meta.name }}-admin"
+    name: "{{ meta.name }}-ingress-controller"
+  - api_version: v1
+    kind: Service
+    namespace: "{{ meta.name }}-admin"
+    name: "{{ meta.name }}-ingress-controller-metrics"
+  - api_version: apps/v1
+    kind: Deployment
+    namespace: "{{ meta.name }}-admin"
+    name: "{{ meta.name }}-ingress-default-backend"
+  - api_version: v1
+    kind: Service
+    namespace: "{{ meta.name }}-admin"
+    name: "{{ meta.name }}-ingress-default-backend"
+  when:
+  - not dryrun
+  - ingressService is defined
+  - ingressService.resources is defined
+  - ingressService.resources[0] is defined
+  - ingressService.resources[0].metadata is defined
+  - ingressService.resources[0].metadata.labels is defined
+  - ingressService.resources[0].metadata.labels.chart is defined
+  - ingressService.resources[0].metadata.labels.chart == "ingress-1.34.2"
+
+- name: Add upgrade status marker
+  k8s_status:
+    api_version: miscscripts.pnnl.gov/v1beta1
+    kind: TenantNamespace
+    name: "{{ meta.name }}"
+    namespace: "{{ meta.namespace }}"
+    status:
+      ingressNginxUpgradeComplete: true
+  when:
+  - not dryrun
+  - ingressService is defined
+  - ingressService.resources is defined
+  - ingressService.resources[0] is defined
+  - ingressService.resources[0].metadata is defined
+  - ingressService.resources[0].metadata.labels is defined
+  - ingressService.resources[0].metadata.labels["helm.sh/chart"] is defined
+  - ingressService.resources[0].metadata.labels["helm.sh/chart"] == "ingress-3.23.0"
+
```

Comment:
We should consider the case, where ingress is disabled.

Reply:
Disabled ingress is now handled when looking up the ingress service, earlier in this file.